### PR TITLE
[webgpu] Add op StringNGrams

### PR DESCRIPTION
**Pull request by axinging**
Note, this called backend.readSync.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5118)

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7  @haoyunfeix @xhcao @gyagp PTAL.
Please note, this called backend.readSync.

File: tfjs-backend-webgpu/src/kernels/StringNGrams.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  if (data.dtype !== 'string') {
+    throw new Error('Data must be of datatype string');
+  }
+  if (dataSplits.dtype !== 'int32') {
+    throw new Error('Data splits must be of datatype int32');
+  }
+  if (data.shape.length !== 1) {
+    throw new Error(`Data must be a vector, saw: ${data.shape}`);
+  }
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/StringNGrams.ts:47]: L39-L47 are repeated verified on cpu/webgl/webgpu. Should this part be done in tfjs-core?

@pyu10055 @lina128 @jinjingforever What's your opinion? Also not only to this PR, there are many files. They are almost same in webgl/webgpu expect for the backendName, like this one. Do you have any suggestion to reduce the duplication?

**Comment by qjia7** [comment]
> Please note, this called backend.readSync.

I suppose currently all string impl is on cpu. There is no gpu impl.

**Review by pyu10055 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5118)* status: 0 of 1 approvals obtained (waiting on @jinjingforever and @lina128)

___
*[tfjs-backend-webgpu/src/kernels/StringNGrams.ts, line 47 at r1](https://reviewable.io/reviews/tensorflow/tfjs/5118#-MaWrBSqheDNLq8ruFUS-r1-47:-MaZSB8i8Q25QMIHuAGt:bqhljdj) ([raw file](https://github.com/tensorflow/tfjs/blob/02dd2f2dbcc91fcf448e4f12f791b2f8b4ce465b/tfjs-backend-webgpu/src/kernels/StringNGrams.ts#L47)):*
<details><summary><i>Previously, qjia7 (Jiajia Qin) wrote…</i></summary><blockquote>

L39-L47 are repeated verified on cpu/webgl/webgpu. Should this part be done in tfjs-core?

@pyu10055 @lina128 @jinjingforever What's your opinion? Also not only to this PR, there are many files. They are almost same in webgl/webgpu expect for the backendName, like this one. Do you have any suggestion to reduce the duplication?
</blockquote></details>

there are some duplication on the data validation, @lina128 will it be ok to remove this check if tf-core API has already done that?

**Review by lina128 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5118)* status: 0 of 1 approvals obtained (waiting on @jinjingforever)

___
*[tfjs-backend-webgpu/src/kernels/StringNGrams.ts, line 47 at r1](https://reviewable.io/reviews/tensorflow/tfjs/5118#-MaWrBSqheDNLq8ruFUS-r1-47:-MaZXj9ID_s5yFgAoZQ4:b-gc5ppf) ([raw file](https://github.com/tensorflow/tfjs/blob/02dd2f2dbcc91fcf448e4f12f791b2f8b4ce465b/tfjs-backend-webgpu/src/kernels/StringNGrams.ts#L47)):*
<details><summary><i>Previously, pyu10055 (Ping Yu) wrote…</i></summary><blockquote>

there are some duplication on the data validation, @lina128 will it be ok to remove this check if tf-core API has already done that?
</blockquote></details>

Yes, validation should be at op level.

**Review by qjia7 — APPROVED** [review]
LGTM.
Leave it to @pyu10055 @lina128 @jinjingforever to merge since the changes are not limited in webgpu.

**Comment by axinging** [comment]
@pyu10055 @lina128 @jinjingforever, PTAL

**Review by lina128 — APPROVED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5118#-:-Mb9-whvAXRUmrDCvnLP:bb6ywmx)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @jinjingforever)

**Pull request merged**